### PR TITLE
Leap Grinder + Burmy Hunter

### PR DESCRIPTION
**Pull request by PhobosLegacy**
- Added a Leap Grinder program (currently there's only the Burmy path)
- Renamed "Shiny Tree Hunter" to "Burmy Hunter" (since there's no point on adding more mons here)
- Refactored the Burmy Hunter to use the base code from Leap Grinder + Enroute Shiny Detection

### CODE REVIEW

File: SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_FroslassFinder.cpp [hunk 1]
```diff
@@ -43,18 +43,17 @@ FroslassFinder::FroslassFinder(const FroslassFinder_Descriptor& descriptor)
...
     , SHINY_DETECTED_DESTINATION(
         "Destination Shiny Action",
-        "This applies if you are near the Froslass.",
+        "This applies if a shiny is detected at or near Froslass.",
         "0 * TICKS_PER_SECOND"
     )
     , NOTIFICATION_STATUS("Status Update", true, false, std::chrono::seconds(3600))
     , NOTIFICATIONS({
         &NOTIFICATION_STATUS,
-        &SHINY_DETECTED_ENROUTE.NOTIFICATIONS,
```
Comment (Mysticial) [inline SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_FroslassFinder.cpp:57]: Keep this line?

File: SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_UnownFinder.cpp [hunk 2]
```diff
@@ -47,21 +45,19 @@ UnownFinder::UnownFinder(const UnownFinder_Descriptor& descriptor)
...
     , SHINY_DETECTED_DESTINATION(
         "Destination Shiny Action",
-        "This applies if you are near the ruins.",
+        "This applies if a shiny is detected inside the ruins.",
         "0 * TICKS_PER_SECOND"
     )
     , NOTIFICATION_STATUS("Status Update", true, false, std::chrono::seconds(3600))
     , NOTIFICATIONS({
         &NOTIFICATION_STATUS,
-        &SHINY_DETECTED_ENROUTE.NOTIFICATIONS,
```
Comment (Mysticial) [inline SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_UnownFinder.cpp:61]: Keep this line.

File: SerialPrograms/Source/PokemonLA/Programs/PokemonLA_TreeActions.cpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+namespace PokemonLA{
+
+
+void setup(ProgramEnvironment& env, ConsoleHandle& console, BotBaseContext& context){
+    goto_camp_from_jubilife(env, console, context, TravelLocations::instance().Fieldlands_Heights);
+    pbf_move_left_joystick(context, 170, 255, 30, 30);
+    change_mount(console, context, MountState::BRAVIARY_ON);
+    pbf_press_button(context, BUTTON_B, (6.35 * TICKS_PER_SECOND), 20);
+    pbf_press_button(context, BUTTON_PLUS, 20, 20);
+    pbf_wait(context, (1 * TICKS_PER_SECOND));
```
Comment (Mysticial) [inline SerialPrograms/Source/PokemonLA/Programs/PokemonLA_TreeActions.cpp:28]: If you want, you can merge each of these waits into the button press above. The last parameter of `pbf_press_button` and family is the wait.

File: SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_FroslassFinder.cpp (hunk 1 above)
Comment (PhobosLegacy) [reply SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_FroslassFinder.cpp]: re-added!

File: SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_UnownFinder.cpp (hunk 2 above)
Comment (PhobosLegacy) [reply SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_UnownFinder.cpp:61]: re-added!

File: SerialPrograms/Source/PokemonLA/Programs/PokemonLA_TreeActions.cpp (hunk 3 above)
Comment (PhobosLegacy) [reply SerialPrograms/Source/PokemonLA/Programs/PokemonLA_TreeActions.cpp]: removed the waits here and in the BurmyFinder too :) thanks!

**Pull request merged**